includes/Stories_Renderer/Renderer.php

```diff
@@ -0,0 +1,489 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Stories_Renderer;
+
+use Google\Web_Stories\Media;
+use Google\Web_Stories\Interfaces\Renderer as RenderingInterface;
+use Google\Web_Stories\Stories;
+use Google\Web_Stories\Story_Post_Type;
+use Iterator;
+
+/**
+ * Renderer class.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ * @implements Iterator<int, \WP_Post>
+ */
+abstract class Renderer implements RenderingInterface, Iterator {
+
+	/**
+	 * Web Stories stylesheet handle.
+	 *
+	 * @var string
+	 */
+	const STYLE_HANDLE = 'web-stories-list-styles';
+
+	/**
+	 * Stories object
+	 *
+	 * @var Stories Stories object
+	 */
+	protected $stories;
+
+	/**
+	 * Story attributes
+	 *
+	 * @var array An array of story attributes.
+	 */
+	protected $attributes = [];
+
+	/**
+	 * Story posts.
+	 *
+	 * @var array An array of story posts.
+	 */
+	protected $story_posts = [];
+
+	/**
+	 * Pointer to iterate over stories.
+	 *
+	 * @var int
+	 */
+	private $position = 0;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Stories $stories Stories instance.
+	 */
+	public function __construct( Stories $stories ) {
+
+		$this->stories    = $stories;
+		$this->attributes = $this->stories->get_story_attributes();
+	}
+
+	/**
+	 * Output markup for amp stories.
+	 *
+	 * @return string
+	 */
+	abstract public function render();
+
+	/**
+	 * Retrieve current story.
+	 *
+	 * @return mixed|void
+	 */
+	public function current() {
+		return $this->story_posts[ $this->position ];
+	}
+
+	/**
+	 * Retrieve next story.
+	 *
+	 * @retrun void
+	 */
+	public function next() {
+		++ $this->position;
+	}
+
+	/**
+	 * Retrieve the key for current node in list.
+	 *
+	 * @return bool|float|int|string|void|null
+	 */
+	public function key() {
+		return $this->position;
+	}
+
+	/**
+	 * Check if current position is valid.
+	 *
+	 * @return bool|void
+	 */
+	public function valid() {
+		return isset( $this->story_posts[ $this->position ] );
+	}
+
+	/**
+	 * Reset pointer to start of the list.
+	 *
+	 * @return void
+	 */
+	public function rewind() {
+		$this->position = 0;
+	}
+
+	/**
+	 * Perform initial setup for object.
+	 *
+	 * @return void
+	 */
+	public function init() {
+
+		$this->story_posts = $this->stories->get_stories();
+	}
+
+	/**
+	 * Initializes renderer functionality.
+	 *
+	 * @return void
+	 */
+	public function assets() {
+
+		wp_enqueue_style(
+			self::STYLE_HANDLE,
+			WEBSTORIES_PLUGIN_DIR_URL . 'includes/assets/stories.css',
+			[],
+			'v0'
+		);
+	}
+
+	/**
+	 * Determine whether the current request is for an AMP page.
+	 *
+	 * @return boolean
+	 */
+	public function is_amp_request() {
+
+		$amp_is_request  = ( function_exists( 'amp_is_request' ) && amp_is_request() );
+		$is_amp_endpoint = ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() );
+
+		return ( $amp_is_request || $is_amp_endpoint );
+	}
```

Comment:
We don't need to check these functions, exist. They will always exist. https://github.com/google/web-stories-wp/blob/1431b73eb81051e98ddf60cedfcebce196cbd8ae/includes/compat/amp.php#L29-L59